odxtools/diagcodedtypes.py

```diff
@@ -519,14 +551,14 @@ def convert_bytes_to_internal(self, decode_state: DecodeState, bit_position: int
             # if isinstance(param_value.parameter, LengthKeyParameter) would be prettier,
             # but leads to cyclic import...
             if (parameter.parameter_type == "LENGTH-KEY" and
-                    parameter.odx_id == self.length_key_id  # type: ignore
+                    parameter.odx_id == self.length_key.odx_id  # type: ignore
```

Comment:
type ignore still needed?

Reply:
unfortunately it is still required:
```
> mypy --ignore-missing-imports odxtools examples/ tests/
odxtools/diagcodedtypes.py:554: error: "Parameter" has no attribute "odx_id"  [attr-defined]
Found 1 error in 1 file (checked 110 source files)
```

this is due to the fact that the base class for parameters  does not exhibit an ID. (only some do, amongst them `LengthKeyParameter`.)

We should only ignore the `attr-defined` error here though. [Fixed](40dfba66b057779aeb45fa4dd0496f14d447ef6b), thanks!